extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -34,23 +33,14 @@ import com.mapbox.maps.logW
  * @see [The online documentation](https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/#geojson)
  *
  */
-class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
+class GeoJsonSource private constructor(builder: Builder) : Source(builder.sourceId) {
```

Comment:
I'm having second thoughts about other Sources than GeoJsonSource. While for this one it's definitely to call `GeoJsonSource(builder)` on the user end other Sources just return `Source(builder)` in their `Builder.build()` definition, e.g. https://github.com/mapbox/mapbox-maps-android/blob/f6bd9832e16405742af74faacfb7f817a9ada03c/extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/ImageSource.kt#L140. 
I can imagine there might be a small change someone is using those constructors. To reduce risk of unneeded runtime crashes could we maybe only make private constructor for the GeoJsonSource? And just deprecate rest of Sources?

Reply:
I think all the Souce classes should have a consistent API

Reply:
I see your point about consistency. At the same time it still concerns me that we introduce potential crash for the users who might use sources with perfectly working constructors of Image/Raster/RasterDem/Vector sources. While for GeoJsonSource this constructor is buggy, other sources are fine.

Reply:
Actually I've just got another idea - use Deprecated annotation with level ERROR - this way it'll remain binary but not  source compatible, thus forcing to use another one @paulrenn67

Reply:
I think it make sense to deprecate and use another one, so we don't break user's code. 
Or we can treat it as a breaking change?

Reply:
Hmmm ok, we can go with the Deprecation approach for now, and remove in v11

Reply:
@yunikkk @ank27 could you please review again, constructors now made `Deprecated`. Had to use `DeprecationLevel.WARNING` though, otherwise our own `Builder.build `gets compile error.

Reply:
I think we might overcomplicated things here, I'd consider it as a bug fix, using constructor with Builder inside it is clearly a misuse to me, and user should fix their code if they were using it that way. wdyt @yunikkk @ank27 @paulrenn67 ?

Reply:
@pengdev it's still better to break only source compatibility in my opinion. Otherwise it's binary breaking and if maps minor version is updated - they'll get a crash. 
At the same time clearly wrong behaviour would be to create GeoJsonSource using constructor with builder, other sources may be perfectly fine built using it.

Reply:
> Had to use DeprecationLevel.WARNING though, otherwise our own Builder.build gets compile error.

@paulrenn67 could you check with  `@Suppress("DEPRECATION_ERROR")` ? Seems it can ignore even the error

Reply:
Ah maybe DEPRECATION_ERROR will do the trick.

Reply:
@paulrenn67 I don't think it's worth the hassle, as we are going to remove the private constructor in v11 (and we don't recommend to use constructor in v10 from the beginning).
Especially these classes are generated and is hard to maintain separately for GeoJsonSource from other sources.
I'm fine with deprecation if you think it's necessary, please avoid complex logic as we are going to v11 soon.

Reply:
@pengdev the issue is that @yunikkk raised the question of whether we should change anything at all.  So either we implement the ticket to make constructors private, or we can just discard this PR and mark the ticket as "won't do".  I think "won't do" makes more sense now if we are going to make bigger changes in V11.

Reply:
> @yunikkk raised the question of whether we should change anything at all.

My preferred solution is to use Deprecated with the Error level, this way we 
- forbid compiling if wrong constructor is used, forcing users to change 
- avoid crashes if maps version was bumped implicitly
The only problem could be that compiler won't allow ourselves to use constructor but if that's hacked with `@Suppress("DEPRECATION_ERROR")` I think there's no more flaws here?

Reply:
Constructor changed to 
```
class GeoJsonSource : Source {
  @Deprecated("Use builder instead", level = DeprecationLevel.ERROR)
  constructor(builder: Builder) : super(builder.sourceId) {
```